src/GitHub.VisualStudio/Helpers/ActiveDocumentSnapshot.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.TextManager.Interop;
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+
+namespace GitHub.VisualStudio
+{
+    [Export(typeof(IActiveDocumentSnapshot))]
+    [PartCreationPolicy(CreationPolicy.NonShared)]
+    class ActiveDocumentSnapshot : IActiveDocumentSnapshot
```

Comment:
It works via reflection, so it doesn't actually care about the visibility of the implementation. But it should be public, yes, for consistency.